server/src/main/webapp/WEB-INF/rails/webpack/views/pages/pipelines/spec/material_editor_spec.tsx

```diff
@@ -148,4 +148,25 @@ describe("AddPipeline: Material Editor", () => {
     expect(helper.byTestId('form-field-input-material-type')).toBeDisabled();
     expect(helper.byTestId('form-field-input-repository-url')).not.toBeDisabled();
   });
+
+  it('should disable only scm material type options when `disableScmMaterials` is set to true', () => {
+    helper.mount(() => <MaterialEditor material={material} showExtraMaterials={true} disableScmMaterials={true}/>);
+
+    const materialTypeSelection = helper.byTestId('form-field-input-material-type');
+    expect(materialTypeSelection).not.toBeDisabled();
+
+    const disabled: string[] = [];
+    const enabled: string[]  = [];
+    helper.qa('option', materialTypeSelection)
+          .forEach((option) => {
+            if (option.hasAttribute('disabled')) {
+              disabled.push(option.textContent!);
+            } else {
+              enabled.push(option.textContent!);
+            }
+          });
+
+    expect(enabled).toEqual(["Another Pipeline", "Package"]);
+    expect(disabled).toEqual(["Git", "Mercurial", "Subversion", "Perforce", "Team Foundation Server", "SCM"]);
```

Comment:
Generally, we should avoid writing loops and logic in the tests. Or else we will need to write a test for the test too :)

Here, can we write multiple assertions, one for each material type?

Reply:
Why would you write a test for looping on a list of `option`? There is no logic here.

Reply:
The test includes a loop and an a condition. Isn't that logic?

I mean, its just a good practice to keep the test simple, which reduces the possibility of bugs in the tests itself.